commands.go

```diff
@@ -1,8 +1,24 @@
 package raft
 
+// VersionInfo is a common sub-structure used to pass along
+// protocol version information. For older Raft implementations before
+// versioning was added this will default to protocol version 0.
+type VersionInfo struct {
```

Comment:
If we added cluster IDs (we should someday), would we want them in this struct? I wonder if this should be named something to the effect of "Common".